File: backend/src/training/sm2_anki/recall.py
from enum import Enum


class Recall(int, Enum):
    FORGET = -2
    SUSPEND = -1
    UNKNOWN = 0
    BAD = 1
    GOOD = 2
    EASY = 3
   
RECALL_INTERVALS = {
    0: 1,
    1: 1.2
}

RECALL_EASINESS_FACTORS = {
    0: 0.8,
    1: 0.85,
    2: 1.0,
    3: 1.15
}

File: frontend/shared/src/training/recall.js
export class Recall { // TODO maybe change this stuff to type script? https://www.freecodecamp.org/news/how-to-add-typescript-to-a-javascript-project/
    static FORGET = Symbol("FORGET")
    static SUSPEND = Symbol("SUSPEND")
    static UNKNOWN = Symbol("UNKNOWN")
    static BAD = Symbol("BAD")
    static GOOD = Symbol("GOOD")
    static EASY = Symbol("EASY")

    static enumMap = {
        "-2": Recall.FORGET,
        "-1": Recall.SUSPEND,
        "0": Recall.UNKNOWN,
        "1": Recall.BAD,
        "2": Recall.GOOD,
        "3": Recall.EASY
    }

    static valueMap = Object.fromEntries(
        Object
            .entries(Recall.enumMap)
            .map(([key, value]) => [value, key])
    )

    static fromValue(value) {
        return Recall.enumMap[value]
    }

    static toValue(symbol) {
        return parseInt(Recall.valueMap[symbol])
    }
}